app/CommandExecutor.php

```diff
@@ -13,6 +13,23 @@ class CommandExecutor
     /** @var array $commands */
     protected $commands = [];
 
+    /** @var string $outputFileName */
+    protected $outputFileName = '';
+
+    /** @var string $errorFileName */
+    protected $errorFileName = '';
```

Comment:
It is possible that within one single command execution the underlying program could output to STDERR and STDOUT alternately... I guess we could keep one single file for the whole outputBuffer, so the sequence between these two outputs does not become puzzling to understand.